packages/py-moose-lib/moose_lib/dmv2/materialized_view.py

```diff
@@ -62,25 +70,38 @@ class MaterializedView(SqlResource, BaseTypedResource, Generic[T]):
     def __init__(
             self,
             options: MaterializedViewOptions,
+            target_table: Optional[OlapTable[T]] = None,
             **kwargs
     ):
         self._set_type(options.materialized_view_name, self._get_type(kwargs))
 
+        # Resolve target table from options
+        if target_table:
+            self.target_table = target_table
+            if self._t != target_table._t:
+                raise ValueError("Target table must have the same type as the materialized view")
```

Comment:
is this a runtime error for the Table concrete type not matching the concrete type of the MaterailziedView?